### ISSUE TITLE
Use short-circuiting for setUserId validation

### ISSUE THREAD

**Issue by stevebrambilla** [amplitude/Amplitude-iOS#37]
When `AMPLITUDE_DEBUG` is set the current `-setUserId` validation incorrectly logs an error when `nil` is sent as the `userId`:

```objc
if (!([self isArgument:userId validType:[NSString class] methodName:@"setUserId:"] || userId == nil)) {
    return;
}
```

Would recommend testing for `nil` first to avoid the erroneous log given that `nil` is now a valid value:

```objc
BOOL validUserId = userId == nil || [self isArgument:userId validType:[NSString class] methodName:@"setUserId:"];
if (!validUserId) {
    return;
}
```

**Comment by djih** [comment]
Hi Steve, 

Thanks for the suggestion. We will roll it into the next release.

**Comment by djih** [comment]
Released with v3.0.1
